pages/profile/[id].tsx

```diff
@@ -92,21 +97,14 @@ const ProfilePage: NextPage<Props> = ({ user }: Props) => {
   );
 };
 
-export const getServerSideProps: GetServerSideProps<Props> = async ({ req, query }) => {
-  const { origin } = absoluteUrl(req);
-
+ProfilePage.getInitialProps = async ({ req, res, query }: NextPageContext): Promise<Props> => {
   try {
-    const config: RawAxiosRequestConfig = req ? {
-      withCredentials: true,
-      headers: {
-        cookie: req.headers.cookie,
-      },
-    } : {};
-
-    const { data: user } = await axios.get<UserProfile>(`${origin}/api/users/${query.id}`, config);
+    const axios = createAxiosInstance(req);
+    const { data: user } = await axios.get<UserProfile>(`/api/users/${query.id}`);
 
-    return { props: { user } };
+    return { user };
   } catch (error) {
+    res.statusCode = error.response.status;
     return { notFound: true };
```

Comment:
should we instead return a null user if not found?  that way we can basically use the truthiness of `user` as whether the user was found or not, and we wouldn't need the optional chaining e.g. `user?.website`

Reply:
Honestly, the optional chaining should never happen because it'll show the 404 template instead.

I just put the optional chaining there while debugging/manual testing and left them because they don't hurt anything. They can be removed if we'd prefer that.

Reply:
yeah, i guess what i'm saying is that if we use `user` as the proxy for `notFound`, we can do something like this:

```js
if (!user) {
  return <NotFound />
}

// user guaranteed to be truthy here
return (
  <div>{user.username}</div>
)
```

Reply:
Ahh, yea. We can do that. I just kept it the same as what `getServerSideProps` expected.

Reply:
Actually, thoughts on passing the actual `statusCode`? Since errors other than just 404 may occur.

```js
if (statusCode !== 200) {
  return <ErrorPage statusCode={statusCode}/>;
}
```

We can still pass a null `user`...
```js
getInitialProps... {
  const props: Props = {
    user: null,
    statusCode: 200,
  };

  try {} catch() {}

  return props;
}
```

or default it to `null` in the params...
```js
const ProfilePage: NextPage<Props> = ({ user = null, statusCode }: Props) => {
...
```

Also, did you still want the optional chaining removed?